cpp/src/io/parquet/page_data.cu

```diff
@@ -689,7 +689,8 @@ inline __device__ void gpuOutputInt96Timestamp(volatile page_state_s *s, int src
     nanos <<= 32;
     nanos |= v.x;
     // Convert from Julian day at noon to UTC seconds
-    secs = (v.z - 2440588) *
+    days = static_cast<int32_t>(v.z);
```

Comment:
Why not cast it directly to int64_t ? May be I am unaware of something.

Reply:
v.z is a uint32 -> it needs to go through uint32->int32->int64 for proper sign extension